state/txindex/kv/kv.go

```diff
@@ -539,29 +554,47 @@ LOOP:
 // Keys
 
 func isTagKey(key []byte) bool {
-	return strings.Count(string(key), tagKeySeparator) == 3
+	// This should be always 4 if data is indexed together with event sequences
+	// The check for 3 was added to allow data indexed before (w/o the event number)
+	// to be retrieved. Note that this is no the cleanest solution, we should probably
+	// redesign this part.
+	numTags := strings.Count(string(key), tagKeySeparator)
+	return numTags == 4 || numTags == 3
```

Comment:
As stated this is potentially a hack. I believe though that the initial intuition here was that the tag keys (which I think are the keys in the query cannot be less than 3 items (key + height + event_type). Thus i deemed it acceptable that it is 3 or 4 to support old versions of the indexer. But people can disagree and we can forego adding event sequencing to the tx indexer.

Reply:
Well, your comment is better than what I see on the left (no comment, so reader is forced to guess).
Also, consider removing the reference to this being "unclean", in any case, it seems to be the same solution as was before.